### DISCUSSION TITLE
RLS policy results in 404

### DISCUSSION THREAD

**Discussion by mrgoldair** [supabase/supabase#14739] [Questions] [answered]
I have an API call

```
.from('logbook')
.insert({
  stage_id,
  minutes,
  billable,
  category_id,
  notes,
  user_id })
.select();
```
which is based on a simple insert policy `(auth.uid() = user_id)`. But unless I manually specify `user_id` in the list of values to `.insert()`, the API call will return `403 - forbidden` on me. Doesn't the client framework have all these details, why do I have to manually pass the `user_id` back?

**Comment by GaryAustin1** [comment]
You should have the user_id column default to auth.uid() in the table.   Then you don’t need to send it and the RLS makes sure it won’t change.

**Reply by GaryAustin1** [reply, marked as answer]
You should reply to comments not start new answers.

When you do the insert, you would not put user_id in that.  The default value for the column will get used automatically if you don't supply.  Note though if you do provide it then it overrides the default.  And your RLS  prevents the user from sending a fake UUID.
The shows setting the default in the UI.
![image](https://github.com/supabase/supabase/assets/54564956/2a65fcb3-3ec7-43a0-a008-cb6367d51135)

**Reply by mrgoldair** [reply]
Not sure why it did that, all I did was reply via email.

That all makes sense, thanks again.

**Comment by mrgoldair** [comment]
Whoa, okay thanks.

How exactly does adding `auth.uid()` as a default save me from sending it in the request?
